File: microservices/Models/Session.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EventManagementSystem.Models
{
    [Table("Sessions", Schema = "EventManagement")]
    public class Session
    {
        [Key]
        public int session_id { get; set; }

        public int event_id { get; set; }

        [Required]
        [StringLength(300)]
        public string title { get; set; } = string.Empty;

        public string? description { get; set; }

        public DateTime start_time { get; set; }
        public DateTime? end_time { get; set; }

        public int? speaker_id { get; set; }
        public int created_by { get; set; }
    }
}

File: front-end/src/models/session.ts
// src/models/session.ts

export interface SessionResponseDto {
  session_id: number;
  event_id: number;
  title: string;
  description: string;
  start_time: string; // ISO 8601 format: "2025-01-10T10:00:00"
  end_time: string;   // ISO 8601 format: "2025-01-10T11:00:00"
  speaker_id: number;
  created_by: number; // User ID of session creator
}

// Frontend interface for displaying sessions
export interface SessionForDisplay extends SessionResponseDto {
  id: string; // Convert session_id to string
  day?: number; // Calculated from start_time
  formattedStartTime?: string; // e.g., "10:00 AM"
  formattedEndTime?: string;   // e.g., "11:00 AM"
  speakerName?: string; // Fetch from Users API or mock
  room?: string;        // Not in backend response yet
  attendees?: number;   // Not in backend response yet
  type?: string;        // Not in backend response yet
}